Add text on matlab parpool bug

wwarriner opened the pull request:
# Pull Request

<!-- PLEASE READ THE COMMENTS BELOW -->

## Overview

<!-- Briefly summarize the proposed changes -->

See title and read added text for more info on what this is about. The short version is: old versions of Matlab had a bug in parpool that causes problems with some workflows. Generally we recommend researchers use the latest Matlab, but that isn't always possible. So, I wrote a workaround. 

## Proposed Changes

<!-- Provide specific details of what is changing -->

## Related Issues

Fixes #531

Premas reviewed (commented):
Looks good to me. I think it would be helpful to refer to this section wherever we discuss MATLAB. For instance, 
https://docs.rc.uab.edu/cheaha/open_ondemand/ood_matlab/
https://docs.rc.uab.edu/cheaha/software/modules/#pre-installed-modules

wwarriner commented:
FYI this is on the same page as the `pre-installed-modules` section.

I added a reference to the OOD Matlab page. I also added an admonition here: http://127.0.0.1:8000/cheaha/slurm/slurm_tutorial/#example-5-multithreaded-or-multicore-job. That tutorial uses `parpool`.

Premas reviewed (approved):
Thanks!
